src/probe_modules/module_dns.c

```diff
@@ -88,10 +86,15 @@ static uint16_t *dns_packet_lens; // Not including udp header
 static uint16_t *qname_lens;
 static char **qnames;
 static uint16_t *qtypes;
-static uint8_t *rdbits;
-static int num_questions = 0;
+static int num_questions = 0; // How many DNS questions to query. Note: There's a requirement that probes is a multiple of DNS questions
+// necessary to null-terminate these since strtrk_r can take multiple delimitors as a char*, and since these are contiguous in memory,
+// they were being used jointly when the intention is to use only one at a time.
+static const char* probe_arg_delimitor = ";\0";
+static const char* domain_qtype_delimitor = ",\0";
+static const char* rn_delimitor = ":\0";
 
-const char *qopts_rn = "rn";
+static uint8_t *rdbits;
+const char *qopts_rn = "nr"; // used in query to disable recursion bit in DNS header
```

Comment:
Updated this, realized that we're using `nr` in the `--help` text, and I think it makes more sense as an abbreviation for `no recursion`